projects/packages/sync/src/modules/class-search.php

```diff
@@ -0,0 +1,1793 @@
+<?php
+/**
+ * Configuration lists for Jetpack Search Fields
+ *
+ * Post Meta: list of post meta keys that are available in the index
+ *   and how they are configured.
+ *
+ * Custom Taxonomy: list of custom taxonomies that are indexed.
+ *
+ * The reason we need an allowed list is that Elasticsearch runs into scaling problems
+ * with more than 200k-ish total fields. The barrier to adding new fields is low,
+ * just open a PR.
+ *
+ * Although the comments indicate specific plugins, you don't need to be running
+ * that plugin for the indexing to work. The metakey just has to match.
+ *
+ * If you need a new meta key or taxonomy also consider using:
+ *   jetpack-search-meta0 - jetpack-search-meta9
+ *   jetpack-search-tag0 - jetpack-search-tag9
+ *
+ * @package automattic/jetpack-sync
+ */
+
+namespace Automattic\Jetpack\Sync\Modules;
+
+/**
+ * Class to handle sync for Jetpack Search.
+ */
+class Search extends Module {
+
+	/**
+	 * Sync module name.
+	 *
+	 * @access public
+	 *
+	 * @return string
+	 */
+	public function name() {
+		return 'search';
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		// Post meta whitelists.
+		add_filter( 'jetpack_sync_post_meta_whitelist', array( $this, 'add_search_post_meta_whitelist' ), 10 );
```

Comment:
I'd suggest we gate this addition to only when Instant Search is active. This has the potential to increase the number of actions flowing to WP.com and we should be responsible to only send if the site is using Instant Search.

Reply:
also we'd need a test on this conditional as well.

Search not active -> these meta are not synced
Search active -> these meta are synced.

--

Yes this will lead to concerns that when users purchase Instant Search the Cache site doesn't have all necessary data, however that already is a concern and we trigger a Full Sync to resolve. This will be further mitigated by rollout of Fixes and so I recommend we maintain not syncing these fields unless Instant Search is active/enabled.